# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

<div class="span5 alert alert-info">
### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

****
</div>

In [46]:
import pandas as pd
import numpy as np
from scipy import stats


%matplotlib inline

In [47]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [48]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [58]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [59]:
#Population samples
n1 = len(data[data.race=='w'])
n2 = len(data[data.race=='b'])
n1,n2

(2435, 2435)

## 1. What test is appropriate for this problem? Does CLT apply?

We can translate this problem into a comparison of two population means. Our populations are the callbacks of white-sounding and black-sounding resumes. Our objective is to use the information from the samples provided in order to estimate the difference in the means of the two populations. We want to know if the mean of the population of callbacks of white-sounding resules is higher than the mean of the black-sounding resumes.
Since the samples are large (2435 samples each) and independent(identical résumés were randomly assigned to black-sounding or white-sounding names), the CLT applies, and therefore, the z-test is an appropriate significance test for this problem.


## 2. What are the null and alternate hypotheses?

**Null hypotesis**: (all equivalent)
- race does not have a significant impact on the rate of callbacks
- there is no significant difference between mean callbacks for white and black applicants  
- mean_callbacks_white = mean_callbacks_black
- The null (and alternative hypotheses) can be also expressed in terms of the difference of the two population means:
- **mean_callbacks_white - mean_callbacks_black = 0** 

**Alternative hypothesis**: 
- race has a significant impact on the rate of callbacks
- mean callbacks for white aplicants is significantly higher than for black applicants  
- mean_callbacks_white > mean_callbacks_black 
- **mean_callbacks_white - mean_callbacks_black > 0**


In [60]:
mean_callbacks_white = (data[data.race=='w'].call).mean()
mean_callbacks_white

0.09650924056768417

In [61]:
mean_callbacks_black = (data[data.race=='b'].call).mean()
mean_callbacks_black


0.0644763857126236

In [62]:
sigma1 = (data[data.race=='w'].call).std()
sigma2 = (data[data.race=='b'].call).std()

In [63]:
#z-test

import math
z = (mean_callbacks_white - mean_callbacks_black) / math.sqrt((math.pow(sigma1,2)/n1) + (math.pow(sigma2,2)/n2))
z

4.114738324427747

Since z = 4.1147 > 1.645, we can reject the null hypothesis 
Therefore, we can conclude that there is a significant difference in callback success according to applicant race.

## 3.Compute margin of error, confidence interval, and p-value.

In [64]:
#Margin of error
#Significance level 0.05--> z = 1.645
#margin of error (ME): ME = critical value * standard error of the mean (mean difference in this case)
#standard_error = math.sqrt((math.pow(sigma1,2)/n1) + (math.pow(sigma2,2)/n2))
ME = 1.645 * math.sqrt((math.pow(sigma1,2)/n1) + (math.pow(sigma2,2)/n2))
ME

0.01280617188309369

In [65]:
#Confidence interval

mean_diff = mean_callbacks_white - mean_callbacks_black
confidence_interval = (mean_diff-ME,mean_diff+ME)

#the 95% confidence interval for the mean of the difference is: 
confidence_interval

(0.019226682971966887, 0.044839026738154264)

**p-value**

The 95% confidence interval of the mean of the difference does not include the zero.
Thus, we can conclude that the the related p-value will be less or equal than 0.05.

## 4.Write a story describing the statistical significance in the context or the original problem.
The aim of this study is to find wether the race has an impact on CV callback success. 
This problem can be translated into a difference of population means problem.
The aim is to find wether the difference of the means of both populations is zero (no difference)
or higher than zero (more success for white-sounding CVs).
With this aim, we have estimated the 95% confidence interval of the difference of the means, and we have seen
that zero does not lie within this interval. This means that it is plausible that the the rate of success 
for white-sounding CVs is significantly higher than for black-sounding CVs.

## 5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

The analysis shown here demonstrates that race may be an important factor in callback success.
However this does not necesarily mean that race is the most important factor. 
There might be other factors as gender, nationality, ... that could also have 
an impact on the callbacks. 
We could include the impact of such factors to our study in order to show wether they are more or less relevant than the race. 